### PR TITLE
Development

### PR DESCRIPTION
**Pull request by mfazza**
# What it is
Lightweight functionality that allows a synapse's project folder structure to be created through configuration files in JSON format.

# What is inside
- synapse builder module
- tests
- configuration files
- some setup files

### CODE REVIEW

File: src/synapsebuilder.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+'''
+createProjectStructure function: takes in a path and executes the
+following flow:
+        1 - Attempt to read configuration file.
+        2 - Attempt to login with Synapse Credentials
+        3 - Create project
+        4 - Attempt to traverse the folder structure while creating folders
+'''
+
+
```
Comment (mattfazza) [inline src/synapsebuilder.py:81]: This allows for this script to be imported into other scripts.  The entire functionality is wrapped by the createProjectInSynapse function.

File: src/synapsebuilder.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    syn = synapseclient.Synapse()
+    syn.login(rememberMe=True)
+
+    projectId = createProject(projectConfig['project'])
+
+    traverseFolders(projectConfig, projectId, projectConfig['project'])
+
+    return projectId
+
+
```
Comment (mattfazza) [inline src/synapsebuilder.py:96]: It can also be used from the command line

File: src/synapsebuilder.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    global syn
+    syn = synapseclient.Synapse()
+    syn.login(rememberMe=True)
+
+    projectId = createProject(projectConfig['project'])
+
+    traverseFolders(projectConfig, projectId, projectConfig['project'])
+
+    return projectId
```
Comment (mfazza) [inline src/synapsebuilder.py]: This allows for this script to be imported into other scripts. The entire functionality is wrapped by the createProjectInSynapse function.

File: src/synapsebuilder.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+
+def main():
+    relativePath = Path(__file__).parent / sys.argv[1]
+    createProjectInSynapse(relativePath)
+
+
+if __name__ == "__main__":
+    main()
```
Comment (mfazza) [inline src/synapsebuilder.py:103]: It can also be used from the command line.

File: src/synapsebuilder.py (hunk 3 above)
Comment (mfazza) [reply src/synapsebuilder.py]: If there's any wish to make this part of the client, then the global variable can go away and the "syn" object can be passed into createProjectInSynapse

**Pull request merged**